pkg/common/apply/interface.go

```diff
@@ -43,20 +45,24 @@ func Apply[T runtime.Object](
 	existing, err := getter.Get(requiredAccessor.GetName())
 	if errors.IsNotFound(err) {
 		actual, createErr := client.Create(ctx, required, metav1.CreateOptions{})
-		if errors.IsAlreadyExists(createErr) {
-			return required, false, nil
-		}
-		if createErr == nil {
+		switch {
+		case errors.IsAlreadyExists(createErr):
+			actual, getErr := client.Get(ctx, requiredAccessor.GetName(), metav1.GetOptions{})
```

Comment:
should we add an comment on why we need to get again?

Reply:
Comment added